vtex/faststore
# Fix/items with promotions in cart

## What's the purpose of this pull request?

Add some existing properties in the order form in the validate cart mutation schema to apply business logic at the promotions level from the store frontend.

Correct the behavior that when the order form divides the same SKU by some promotion, it does not correctly unify the values related to it.

## How does it work?

Add price tags object and price definition object of each order form item in validate cart mutation schema.
![image](https://github.com/vtex/faststore/assets/37114506/695b55fe-a6f1-413e-9742-139c7932d479)

Modify join items function to also unify the price tags object and price definition object so that no information is lost.
![image](https://github.com/vtex/faststore/assets/37114506/39c0e2a0-5956-491e-ba1e-e38c3b19b8b5)

## How to test it?

Make a validate cart mutation with products that apply for promotion and that the order form separates them by default. The expected result is that the item returns with all the information related to the promotion and that the values are consistent according to the amount of the product added.
